appeals/web/src/server/lib/mappers/appeal/submappers/s106-obligation-due-date.mapper.js

```diff
@@ -1,30 +1,13 @@
 import { dateISOStringToDisplayDate } from '#lib/dates.js';
+import { textSummaryListItem } from '#lib/mappers/components/text.js';
 
 /** @type {import('../appeal.mapper.js').SubMapper} */
-export const mapS106ObligationDue = ({ appealDetails, currentRoute }) => ({
-	id: 's106-obligation-due-date',
-	display: {
-		summaryListItem: {
-			key: {
-				text: 'S106 obligation due'
-			},
-			value: {
-				html: dateISOStringToDisplayDate(appealDetails.appealTimetable?.s106ObligationDueDate) || ''
-			},
-			actions: {
-				items: [
-					appealDetails.validAt
-						? {
-								text: 'Change',
-								href: `${currentRoute}/appeal-timetables/s106-obligation`,
-								visuallyHiddenText:
-									'The date all case documentation was received and the appeal was valid',
-								attributes: { 'data-cy': 'change-s106-obligation-due-date' }
-						  }
-						: {}
-				]
-			},
-			classes: 'appeal-s106-obligation-due-date'
-		}
-	}
-});
+export const mapS106ObligationDue = ({ appealDetails, currentRoute, userHasUpdateCasePermission }) =>
+	textSummaryListItem({
+		id: 's106-obligation-due-date',
+		text: 'S106 obligation due',
+		value: dateISOStringToDisplayDate(appealDetails.appealTimetable?.s106ObligationDueDate) || '',
```

Comment:
I haven't actually looked at this function so this might be a bad call. Seems like every time this is used we fall back to a string, normally an empty one. Could `dateISOStringToDisplayDate` be made to take a fallback string as an optional argument and fall back to an empty string by default?

Reply:
this is merged already, but will add another PR